### PR TITLE
Creative tab proposal

### PR DESCRIPTION
**Pull request by dshadowwolf**
Proposal for making creative tabsthrough MMDLib a per-mod setup and mostly severing them from any underlying structure in MMDLib. The final part of that is a WIP that is part of why this proposal is not eing suggested fro a direct pull.

1) Remove final ties to any specific naming in MMDLib - this will likely require some architecture changes. (I'm out of ideas on how to make this happen in a maintainable and good manner)
2) The setup for the "Tab Icon Item" should be changed to a Callable<> or similar to match the code and style in MMDLib - I'm not ashamed to say that I've not worked with this in the past and am unsure of how to make it work. (Though... if Callable<> is like a Promise, then... it won't be too hard)

### CODE REVIEW

File: src/main/java/com/mcmoddev/basemetals/init/Items.java [hunk 1]
```diff
@@ -33,42 +36,42 @@ public static void init() {
...
 			material.addNewItem(Names.INGOT, new ItemStack(net.minecraft.init.Items.COAL, 1, 1).getItem());
 
-			createNugget(material);
-			createPowder(material);
-			createSmallPowder(material);
-			FuelRegistry.addFuel(Oredicts.NUGGET_CHARCOAL, 200);
+			createNugget(material, myTabs.itemsTab);
+			createPowder(material, myTabs.itemsTab);
+			createSmallPowder(material, myTabs.itemsTab);
+			FuelRegistry.addFuel(Oredicts.NUGGET+"_charcoal", 200);
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/basemetals/init/Items.java]: No need for the Oredict change, was temporarily broken, now fixed.

File: src/main/java/com/mcmoddev/basemetals/init/Items.java [hunk 2]
```diff
@@ -79,26 +82,26 @@ public static void init() {
...
 			material.addNewItem(Names.INGOT, net.minecraft.init.Items.COAL);
 
-			createNugget(material);
-			createPowder(material);
-			createSmallPowder(material);
-			FuelRegistry.addFuel(Oredicts.NUGGET_COAL, 200);
+			createNugget(material, myTabs.itemsTab);
+			createPowder(material, myTabs.itemsTab);
+			createSmallPowder(material, myTabs.itemsTab);
+			FuelRegistry.addFuel(Oredicts.NUGGET+"_coal", 200);
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/basemetals/init/Items.java]: No need for the Oredict change, was temporarily broken, now fixed.

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 3]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createPlate(MMDMaterial material) {
-		final Block block = createBlock(material, Names.PLATE, BlockMMDPlate.class, Options.enablePlate, true);
-		OreDictionary.registerOre(Oredicts.PLATE + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createPlate(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 4]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createBars(MMDMaterial material) {
-		final Block block = createBlock(material, Names.BARS, BlockMMDBars.class, Options.enableBars, true);
-		OreDictionary.registerOre(Oredicts.BARS + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createBars(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 5]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
-	protected static Block createBlock(MMDMaterial material, boolean glow) {
+	protected static Block createBlock(MMDMaterial material, boolean glow, CreativeTabs tab) {
 		if (material == null) {
 			return null;
 		}
 
 		if ((Options.enableBasics) && (!material.hasBlock(Names.BLOCK))) {
-			final Block block = addBlock(new BlockMMDBlock(material, glow, true), Names.BLOCK, material, ItemGroups.blocksTab);
-			OreDictionary.registerOre(Oredicts.BLOCK + material.getCapitalizedName(), block);
+				addBlock(new BlockMMDBlock(material, glow, true), Names.BLOCK, material, tab);
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 6]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createButton(MMDMaterial material) {
-		final Block block = createBlock(material, Names.BUTTON, BlockMMDButton.class, Options.enableButton, true);
-		OreDictionary.registerOre(Oredicts.BUTTON + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createButton(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 7]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createLever(MMDMaterial material) {
-		final Block block = createBlock(material, Names.LEVER, BlockMMDLever.class, Options.enableLever, true);
-		OreDictionary.registerOre(Oredicts.LEVER + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createLever(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 8]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	/**
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static BlockSlab createSlab(MMDMaterial material) {
-		// oreDict is handled in items
-		return (BlockSlab)createBlock(material, Names.HALFSLAB, BlockMMDHalfSlab.class, Options.enableSlab, true);
+	protected static BlockSlab createSlab(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back (comment)

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 9]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	/**
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static BlockSlab createDoubleSlab(MMDMaterial material) {
-		// oreDict is handled in items
-		return (BlockSlab)createBlock(material, Names.DOUBLESLAB, BlockMMDDoubleSlab.class, Options.enableSlab, true);
+	protected static BlockSlab createDoubleSlab(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back (comment)

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 10]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createStairs(MMDMaterial material) {
-		final Block block = createBlock(material, Names.STAIRS, BlockMMDStairs.class, Options.enableStairs, material.hasBlock(Names.BLOCK));
-		OreDictionary.registerOre(Oredicts.STAIRS + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createStairs(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 11]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createWall(MMDMaterial material) {
-		final Block block = createBlock(material, Names.WALL, BlockMMDWall.class, Options.enableWall, material.hasBlock(Names.BLOCK));
-		OreDictionary.registerOre(Oredicts.WALL + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createWall(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 12]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createOre(MMDMaterial material) {
-		final Block block = createBlock(material, Names.ORE, BlockMMDOre.class, Options.enableBasics, !material.hasBlock(Names.ORE) && material.hasOre());
-		OreDictionary.registerOre(Oredicts.ORE + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createOre(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 13]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createEndOre(MMDMaterial material) {
-		final Block block = createBlock(material, Names.ENDORE, BlockMMDEndOre.class, Options.enableBasics, !material.hasBlock(Names.ENDORE) && material.hasOre());
-		OreDictionary.registerOre(Oredicts.ORE_END + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createEndOre(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 14]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createNetherOre(MMDMaterial material) {
-		final Block block = createBlock(material, Names.NETHERORE, BlockMMDNetherOre.class, Options.enableBasics, !material.hasBlock(Names.NETHERORE) && material.hasOre());
-		OreDictionary.registerOre(Oredicts.ORE_NETHER + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createNetherOre(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 15]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	/**
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static BlockDoor createDoor(MMDMaterial material) {
-		// oreDict is handled in items
-		return (BlockDoor)createBlock(material, Names.DOORBLOCK, BlockMMDDoor.class, Options.enableDoor, true);
+	protected static BlockDoor createDoor(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back (comment)

File: src/main/java/com/mcmoddev/lib/init/Blocks.java [hunk 16]
```diff
@@ -218,233 +221,231 @@ protected static Block createBookshelf(MMDMaterial material, boolean fullBlock)
...
 	 * 
 	 * @param material The material this is made from
+	 * @param tab which creative tab is it on
 	 * @return the block this function created
 	 */
-	protected static Block createTrapDoor(MMDMaterial material) {
-		final Block block = createBlock(material, Names.TRAPDOOR, BlockMMDTrapDoor.class, Options.enableTrapdoor, true);
-		OreDictionary.registerOre(Oredicts.TRAPDOOR + material.getCapitalizedName(), block);
-		return block;
+	protected static Block createTrapDoor(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Blocks.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 17]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this crystal
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createCrystal(MMDMaterial material) {
-		final Item item = createItem(material, Names.CRYSTAL, ItemMMDIngot.class, Options.enableBasics, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.CRYSTAL + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createCrystal(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 18]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this gem
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createGem(MMDMaterial material) {
-		final Item item = createItem(material, Names.GEM, ItemMMDIngot.class, Options.enableBasics, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.GEM + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createGem(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 19]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this ingot
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createIngot(MMDMaterial material) {
-		final Item item = createItem(material, Names.INGOT, ItemMMDIngot.class, Options.enableBasics, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.INGOT + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createIngot(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 20]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this nugget
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createNugget(MMDMaterial material) {
-		final Item item = createItem(material, Names.NUGGET, ItemMMDNugget.class, Options.enableBasics, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.NUGGET + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createNugget(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 21]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this powder
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createPowder(MMDMaterial material) {
-		final Item item = createItem(material, Names.POWDER, ItemMMDPowder.class, Options.enableBasics, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.DUST + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createPowder(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 22]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this blend
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createBlend(MMDMaterial material) {
-		final Item item = createItem(material, Names.BLEND, ItemMMDBlend.class, Options.enableBasics, material.hasBlend(), ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.DUST + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createBlend(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 23]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createRod(MMDMaterial material) {
-		Item item = createItem(material, Names.ROD, ItemMMDRod.class, Options.enableRod, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.ROD + material.getCapitalizedName(), item);
-		OreDictionary.registerOre(Oredicts.STICK + material.getCapitalizedName(), item);
-		OreDictionary.registerOre(Oredicts.ROD, item);
-		return item;
+	protected static Item createRod(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back (missing one entry)

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 24]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * @param material The material base of this gear
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createGear(MMDMaterial material) {
-		Item item = createItem(material, Names.GEAR, ItemMMDGear.class, Options.enableGear, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.GEAR + material.getCapitalizedName(), item);
-		OreDictionary.registerOre(Oredicts.GEAR, item);
-		return item;
+	protected static Item createGear(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back (missing one entry)

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 25]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createArrow(MMDMaterial material) {
-		final Item item = createItem(material, Names.ARROW, ItemMMDArrow.class, Options.enableBowAndArrow, true, ItemGroups.toolsTab);
-		OreDictionary.registerOre(Oredicts.ARROW + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createArrow(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 26]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createBolt(MMDMaterial material) {
-		final Item item = createItem(material, Names.BOLT, ItemMMDBolt.class, Options.enableCrossbowAndBolt, true, ItemGroups.toolsTab);
-		OreDictionary.registerOre(Oredicts.AMMOBOLT, item);
-		return item;
+	protected static Item createBolt(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 27]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createSmallBlend(MMDMaterial material) {
-		final Item item = createItem(material, Names.SMALLBLEND, ItemMMDSmallBlend.class, Options.enableSmallDust, material.hasBlend(), ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.DUST_TINY + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createSmallBlend(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 28]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createSmallPowder(MMDMaterial material) {
-		final Item item = createItem(material, Names.SMALLPOWDER, ItemMMDSmallPowder.class, Options.enableSmallDust, material.hasBlend(), ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.DUST_TINY + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createSmallPowder(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 29]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createShield(MMDMaterial material) {
-		final Item item = createItem(material, Names.SHIELD, ItemMMDShield.class, Options.enableShield, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.SHIELD + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createShield(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 30]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createMekCrystal(MMDMaterial material) {
-		Item item = createItem(material, Names.CRYSTAL, GenericMMDItem.class, Options.enableMekanism, material.getType()!=MaterialType.CRYSTAL, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.CRYSTAL + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createMekCrystal(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 31]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createMekShard(MMDMaterial material) {
-		Item item = createItem(material, Names.SHARD, GenericMMDItem.class, Options.enableMekanism, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.SHARD + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createMekShard(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 32]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createMekClump(MMDMaterial material) {
-		Item item = createItem(material, Names.CLUMP, GenericMMDItem.class, Options.enableMekanism, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.CLUMP + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createMekClump(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 33]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createMekDirtyPowder(MMDMaterial material) {
-		Item item = createItem(material, Names.POWDERDIRTY, GenericMMDItem.class, Options.enableMekanism, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.DUST_DIRTY + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createMekDirtyPowder(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 34]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createCasing(MMDMaterial material) {
-		Item item = createItem(material, Names.CASING, GenericMMDItem.class, Options.enableIC2, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.CASING + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createCasing(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 35]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createDensePlate(MMDMaterial material) {
-		Item item = createItem(material, Names.DENSEPLATE, GenericMMDItem.class, Options.enableIC2, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.PLATE_DENSE + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createDensePlate(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 36]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createCrushed(MMDMaterial material) {
-		Item item = createItem(material, Names.CRUSHED, GenericMMDItem.class, Options.enableIC2, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.CRUSHED + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createCrushed(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 37]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createCrushedPurified(MMDMaterial material) {
-		Item item = createItem(material, Names.CRUSHEDPURIFIED, GenericMMDItem.class, Options.enableIC2, true, ItemGroups.itemsTab);
-		OreDictionary.registerOre(Oredicts.CRUSHED_PURIFIED + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createCrushedPurified(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 38]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * 
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createSlab(MMDMaterial material) {
-		final Item item = createItem(material, Names.SLAB, ItemMMDSlab.class, Options.enableSlab, material.hasBlock(Names.HALFSLAB) && material.hasBlock(Names.DOUBLESLAB), ItemGroups.blocksTab);
-		OreDictionary.registerOre(Oredicts.SLAB + material.getCapitalizedName(), item);
-		return item;
+	protected static Item createSlab(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Needs oredict stuff put back

File: src/main/java/com/mcmoddev/lib/init/Items.java [hunk 39]
```diff
@@ -304,410 +310,422 @@ private static Item createArmorItem(MMDMaterial material, Names name) {
...
 	 * @param material The material base of this item
+	 * @param tab which creative tab it is in
 	 * @return the item this function created
 	 */
-	protected static Item createDoor(MMDMaterial material) {
-		final Item item = createItem(material, Names.DOOR, ItemMMDDoor.class, Options.enableDoor, material.hasBlock(Names.DOORBLOCK), ItemGroups.blocksTab);
-		OreDictionary.registerOre(Oredicts.DOOR + material.getCapitalizedName(), item);
-		OreDictionary.registerOre(Oredicts.DOOR, item);
-		return item;
+	protected static Item createDoor(MMDMaterial material, CreativeTabs tab) {
```
Comment (jriwanek) [inline src/main/java/com/mcmoddev/lib/init/Items.java]: Item it -> Item item
Needs oredict stuff put back

**Review by jriwanek — CHANGES_REQUESTED** [review]

**Comment by adrwa** [comment]
more shit?

**Review by jriwanek — APPROVED** [review]

**Pull request merged**